src/net/libwebsocketsIO.cpp

```diff
@@ -321,6 +321,12 @@ static bool check_public_key(X509_STORE_CTX* ctx)
     EVP_PKEY* evp;
     if ((evp = X509_PUBKEY_get(X509_get_X509_PUBKEY(X509_STORE_CTX_get0_cert(ctx)))))
     {
+        if (EVP_PKEY_id(evp) != EVP_PKEY_RSA)
```

Comment:
This protection should be added to the homologous piece of code in the SDK, `cert_verify_callback()`. It is not currently in use for connections to chat-servers (it used to be, when DNS resolution was entirely done by the SDK and it checked the IP was reachable), but to API servers, and they might be affected by the same issue, so it deserves a patch too

Reply:
Done in PR #2069